state/protocol/badger/state.go

```diff
@@ -155,6 +163,12 @@ func Bootstrap(
 func (state *State) bootstrapSealingSegment(segment *flow.SealingSegment, head *flow.Block) func(tx *transaction.Tx) error {
 	return func(tx *transaction.Tx) error {
 
+		blocksById := make(map[flow.Identifier]*flow.Block, len(segment.Blocks))
```

Comment:
I think we can roll back all the changes to `bootstrapSealingSegment`. 

The additions to `bootstrapSealingSegment` here are to process any `VersionBeacon` service events which appear in the sealing segment. However, as the code mentions, it is quite unlikely that the most recent `VersionBeacon` service event actually happens to appear in the sealing segment. 

Therefore, we need to bootstrap the `VersionBeacon` information separately in the general case, which we do in `boostrapVersionBeacon`. 

Is there any case where `boostrapVersionBeacon` is insufficient on its own? I don't think so, here is my reasoning:

```
A <- B <- C <- D
```

We always initialize an initial `VersionBeacon` value, based on the value for the root block of the snapshot (the highest block: `D`). Therefore, we don't need to worry about separately processing service events which are applied as of `D`. 

Let's suppose there is a service event `S` which is not yet applied as of block `D`. That means that `S` will be sealed either by `D`, or a block later than `D` (after the sealing segment). 

If `S` is sealed by `D`, the `S` will be applied when finalizing `D`'s child. This will be handled by the `Mutator` and does not need a special case in bootstrapping. The same reasoning applies in case `S` is sealed by a block later than `D`.

Reply:
https://github.com/onflow/flow-go/pull/3736#issuecomment-1360449088 - especially the first question. If that's correct I'm happy to remove changes in `bootstrapSealingSegment` - just want to be extra sure